app/controllers/jobseekers/job_applications/self_disclosure_controller.rb

```diff
@@ -0,0 +1,95 @@
+class Jobseekers::JobApplications::SelfDisclosureController < Jobseekers::BaseController
+  include Wicked::Wizard
+
+  before_action :redirect_when_self_disclosure_not_pending, except: :completed
+  before_action :form, except: :completed
+
+  helper_method :job_application
+
+  FORMS = {
+    personal_details: Jobseekers::JobApplications::SelfDisclosure::PersonalDetailsForm,
+    barred_list: Jobseekers::JobApplications::SelfDisclosure::BarredListForm,
+    conduct: Jobseekers::JobApplications::SelfDisclosure::ConductForm,
+    confirmation: Jobseekers::JobApplications::SelfDisclosure::ConfirmationForm,
+  }.freeze
+
+  steps(*FORMS.keys)
+
+  def show
+    render_wizard
+  end
+
+  def update
+    if form.update!
```

Comment:
This pattern is out of sync with the rest of the codebase. if @form.valid? then update is more the style that is used.